criteria/geode/src/org/immutables/criteria/geode/OqlGenerator.java

```diff
@@ -115,6 +113,15 @@ Oql generate(Query query) {
     return new Oql(variables, oql.toString());
   }
 
+  private static boolean isMultiDirectionCollation(Query query) {
+    final long directionCount = query.collations().stream()
+            .map(Collation::direction)
+            .distinct()
+            .limit(Ordering.Direction.values().length)
```

Comment:
Will `distinct()` return at most `Ordering.Direction.values().length` anyway ?

Reply:
yes but this short circuits the count to avoid iterating over all collations